### ISSUE TITLE
Docs: mention to put server-settings into settings component

### ISSUE THREAD

**Issue by ssiccha** [neovim/nvim-lspconfig#1951]
### Requested feature

Could you add to the README.md, that when one wants to pass settings to the LSP server in the `require('lspconfig')[<lsp>].setup` call, one has to wrap that into a `settings` component like so:
```
require('lspconfig')['pylsp'].setup {
  ...,
  settings = {
      pylsp = {
          plugins = {
              ...
          },
      },
  },
}
```
I was trying to configure `extra_paths` for `pylsp` and since I'm a LSP newbie and there are so many parts involved, it took me literally hours to figure this out (I learned a lot in the progress but that's another point).

I saw a settings component being used in another config example, but I didn't make the connection that this is a general pattern. Maybe this behavior is mentioned somewhere, but if that's so I either didn't look right or it might be hidden a bit.

**Comment by justinmk** [comment]
it's mentioned over and over in https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md 

though the formatting is bad in many cases.

**Comment by ssiccha** [comment]
Yes. For example it's not mentioned for `pylsp`, `pyre`, `puppet`, `purescriptls`, `quick_lint_js`, and `r_language_server`. But it *is* present in `pyright`. Which is why I thought it's a setting special to `pyright`.

Maybe one could update the [suggested configuration in the README.md](https://github.com/neovim/nvim-lspconfig#suggested-configuration) to include a table of settings per server like so?

```
-- Use a loop to conveniently call 'setup' on multiple servers and
-- map buffer local keybindings when the language server attaches
local servers = { 'pyright', 'rust_analyzer', 'tsserver' }
-- Put server specific settings here:
local settings = {
  pyright = {},
  rust_analyzer = {},
  tsserver = {},
}
for _, lsp in pairs(servers) do
  require('lspconfig')[lsp].setup {
    on_attach = on_attach,
    flags = {
      -- This will be the default in neovim 0.7+
      debounce_text_changes = 150,
    }
    settings = settings[lsp]
  }
end
```
